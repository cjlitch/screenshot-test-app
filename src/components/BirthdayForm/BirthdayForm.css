.birthday-container {
  margin-top: 100px;
  background-image: url("../../images/constellation.png");
  background-size: 600px;
  background-repeat: no-repeat;
  background-position: top right;
  min-height: 300px;
}

.birthday-form-button {
  padding: 20px;
  background-color: #ddd;
  font-family: 'Spectral SC', serif;
  font-weight: 400;
  font-size: 18px;
  margin-top: 60px;
  display: inline-block;
  text-align: center;
  cursor: pointer;
  border-radius: 3px;
  border-bottom: 1px solid #bbb;
  border-top: 1px solid #ccc;  
  color: #4c4c4c;
  text-decoration: none;
}

.birthday-form-button:hover {
  cursor: pointer;
}

.birthday-form-button:active {
  background-color: #bbb;
  border-top: 1px solid #aaa;
  cursor: pointer;
}

.horoscope {
  text-align: center;
  font-family: 'Spectral SC', serif;
  font-weight: 400;
  color: #4c4c4c;
  margin-top: 100px;
}