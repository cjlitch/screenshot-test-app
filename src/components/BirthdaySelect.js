import React, { Component } from 'react';
import Select from 'react-select';
import 'react-select/dist/react-select.css';

class BirthdaySelect extends Component {

    state = {
        selectedOption: '',
    }
  
    handleChange = (selectedOption) => {
        this.setState({ selectedOption });
    }

    render() {
        return (
            <div>
            <label for="day">Day</label>
            <Select
                name="day"
                value={this.state.selectedOption.value}
                onChange={this.handleChange}
                clearable={false}
                options={[
                    { value: 'one', label: 'One' },
                    { value: 'two', label: 'Two' },
                ]} />
            </div>
        );
    }
}

export default BirthdaySelect;
