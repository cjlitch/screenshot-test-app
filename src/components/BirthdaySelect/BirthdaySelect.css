.birthday-select {
  margin-top: 20px;
  max-width: 200px;
}

.birthday-label {
  font-family: 'Spectral SC', serif;
  font-weight: 400;
}