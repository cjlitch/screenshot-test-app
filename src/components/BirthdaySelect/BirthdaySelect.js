import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Select from 'react-select';
import 'react-select/dist/react-select.css';
import styled from 'styled-components';
import styleguide from '../../styleguide';

export default class BirthdaySelect extends Component {
    static propTypes = {
        label: PropTypes.string.isRequired,
        options: PropTypes.array.isRequired,
        onChange: PropTypes.func.isRequired,
    }

    state = {
        selectedOption: '',
    }
  
    handleChange = (selectedOption) => {
        this.setState({ selectedOption });
        this.props.onChange(selectedOption);
    }

    render() {
        return (
            <StyledSelect data-test="birthday-select">
                <Label htmlFor={this.props.label}>
                    {this.props.label}
                </Label>
                <Select
                    name={this.props.label}
                    value={this.state.selectedOption.value}
                    onChange={this.handleChange}
                    clearable={false}
                    options={this.props.options} />
            </StyledSelect>
        );
    }
}

const Label = styled.label`
    font-family: ${styleguide.fonts.spectral};
    color: ${styleguide.colors.grayTundora};
    font-weight: 400;
`;

const StyledSelect = styled.div`
    margin-top: 20px;
    max-width: 200px;`
;
