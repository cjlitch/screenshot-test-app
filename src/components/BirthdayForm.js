import React, { Component } from 'react';
import BirthdaySelect from './BirthdaySelect';

class BirthdayForm extends Component {
    render() {
        return (
            <div className="container">
                <h2 className="form-title">Get your Cowboy Horoscope now</h2>
                <p className="form-subtitle">Enter the month and day of your birthday</p>
                <BirthdaySelect />
                <BirthdaySelect />
            </div>
        );
    }
}

export default BirthdayForm;
