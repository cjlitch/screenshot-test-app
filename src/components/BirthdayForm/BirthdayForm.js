import React, { Component } from 'react';
import PropTypes from 'prop-types';
import BirthdaySelect from '../BirthdaySelect/BirthdaySelect';
import './BirthdayForm.css';

class BirthdayForm extends Component {
    static propTypes = {
        setMonth: PropTypes.func.isRequired,
        setDay: PropTypes.func.isRequired,
        months: PropTypes.array.isRequired,
        days: PropTypes.array.isRequired,
        revealHoroscope: PropTypes.func.isRequired,
        horoscope: PropTypes.string,
    }

    render() {
        return (
            <div className="container birthday-container">
                <h2 className="form-title">Get your Cowboy Horoscope now</h2>
                <p className="form-subtitle">Enter the month and day of your birthday</p>
                
                <BirthdaySelect
                    label={"Month"}
                    options={this.props.months}
                    onChange={(selected) => this.props.setMonth(selected)} />

                <BirthdaySelect
                    label={"Day"}
                    options={this.props.days}
                    onChange={(selected) => this.props.setDay(selected)} />

                <div
                    className="birthday-form-button"
                    onClick={this.props.revealHoroscope}
                    data-test="reveal-horoscope">
                    Reveal your Cowboy Horoscope
                </div>

                <h1 className="horoscope" data-test="horoscope">{this.props.horoscope}</h1>
            </div>
        );
    }
}

export default BirthdayForm;
