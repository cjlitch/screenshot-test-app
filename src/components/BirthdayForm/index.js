import { connect } from 'react-redux';
import BirthdayForm from './BirthdayForm';
import { setDay, setMonth, revealHoroscope } from '../../actions';

const mapStateToProps = state => {
    return {
        months: state.months,
        days: state.days,
        selectedMonth: state.selectedMonth,
        selectedDay: state.selectedDay,
        horoscope: state.horoscope,
    };
}

const mapDispatchToProps = dispatch => {
    return {
        setDay: day => {
            dispatch(setDay(day));
        },
        setMonth: month => {
            dispatch(setMonth(month));
        },
        revealHoroscope: () => {
            dispatch(revealHoroscope());
        }
    }
}

const BirthdayFormContainer = connect(
    mapStateToProps,
    mapDispatchToProps
)(BirthdayForm)

export default BirthdayFormContainer;