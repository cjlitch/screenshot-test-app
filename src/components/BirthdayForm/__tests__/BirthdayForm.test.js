import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import BirthdayForm from '../BirthdayForm';
import BirthdaySelect from '../../BirthdaySelect/BirthdaySelect';
import renderer from 'react-test-renderer';

const defaultProps = {
    months: [1, 2, 3],
    days: [1, 2, 3],
    setMonth: () => {},
    setDay: () => {},
    revealHoroscope: () => {}
};

describe('BirthdayForm', () => {
    it('has two BirthdaySelects', () => {
        const wrapper = shallow(<BirthdayForm {...defaultProps} />);
        expect(wrapper.find(BirthdaySelect)).toHaveLength(2);
    });

    it('matches snapshot', () => {
        const wrapper = renderer
            .create(<BirthdayForm {...defaultProps} />)
            .toJSON();

        expect(wrapper).toMatchSnapshot();
    });
});


