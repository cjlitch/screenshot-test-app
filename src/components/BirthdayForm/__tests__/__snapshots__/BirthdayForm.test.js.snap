// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BirthdayForm matches snapshot 1`] = `
<div
  className="container birthday-container"
>
  <h2
    className="form-title"
  >
    Get your Cowboy Horoscope now
  </h2>
  <p
    className="form-subtitle"
  >
    Enter the month and day of your birthday
  </p>
  <div
    className="birthday-select"
    data-test="birthday-select"
  >
    <label
      className="birthday-label"
      htmlFor="Month"
    >
      Month
    </label>
    <div
      className="Select is-searchable Select--single"
      style={undefined}
    >
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-2--value"
        >
          <div
            className="Select-placeholder"
          >
            Select...
          </div>
          <div
            className="Select-input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-activedescendant="react-select-2--value"
              aria-describedby={undefined}
              aria-expanded="false"
              aria-haspopup="false"
              aria-label={undefined}
              aria-labelledby={undefined}
              aria-owns=""
              className={undefined}
              id={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              role="combobox"
              style={
                Object {
                  "boxSizing": "content-box",
                  "width": "5px",
                }
              }
              tabIndex={undefined}
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="birthday-select"
    data-test="birthday-select"
  >
    <label
      className="birthday-label"
      htmlFor="Day"
    >
      Day
    </label>
    <div
      className="Select is-searchable Select--single"
      style={undefined}
    >
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-3--value"
        >
          <div
            className="Select-placeholder"
          >
            Select...
          </div>
          <div
            className="Select-input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-activedescendant="react-select-3--value"
              aria-describedby={undefined}
              aria-expanded="false"
              aria-haspopup="false"
              aria-label={undefined}
              aria-labelledby={undefined}
              aria-owns=""
              className={undefined}
              id={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              role="combobox"
              style={
                Object {
                  "boxSizing": "content-box",
                  "width": "5px",
                }
              }
              tabIndex={undefined}
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
  <a
    className="birthday-form-button"
    data-test="reveal-horoscope"
    href="#footer"
    onClick={[Function]}
    onTouchStart={[Function]}
  >
    Reveal your Cowboy Horoscope
  </a>
  <h1
    className="horoscope"
    data-test="horoscope"
  />
</div>
`;
